src/main/java/org/jenkinsci/plugins/parallel_test_executor/SplitStep.java

```diff
@@ -75,13 +76,14 @@ public DescriptorImpl() {
         @StepContextParameter private transient TaskListener listener;
 
         @Override protected List<?> run() throws Exception {
+            FilePath path = this.getContext().get(FilePath.class);
```

Comment:
Note that this will be null if not inside `node`. (The usual case for existing scripts not prepared to take advantage of this feature.)

You may also want to update [this](https://github.com/jenkinsci/parallel-test-executor-plugin/blob/a5910785979c4b40e02e7fc4ebe8ecc69fdc5c54/demo/lib/vars/testInParallel.groovy#L2) to read something like

```groovy
def splits
node {
  prepare()
  splits = splitTests parallelism: parallelism, generateInclusions: true
}
```

and adjust the [documentation](https://github.com/jenkinsci/parallel-test-executor-plugin/blob/a5910785979c4b40e02e7fc4ebe8ecc69fdc5c54/demo/README.md#demo-contents) to note that even the first build (or the first build of a new branch) will have some reasonable default split.

Reply:
Of course. I will add an example how to use it - it should not be obvious for user that this feature does not have sense without node and project source code.

Reply:
Jesee, I am not sure what you meant by this "Was expecting something in ParallelTestExecutorTest demonstrating the effectiveness of the new feature in the context of an actual Pipeline build". Do you wish a test which compares two pipeline jobs execution (one without new feature and one with new feature) in the matter of time?